tests/unit_tests/sql_parse_tests.py

```diff
@@ -525,6 +558,16 @@ def test_extract_tables_reusing_aliases() -> None:
 with q1 as ( select key from q2 where key = '5'),
 q2 as ( select key from src where key = '5')
 select * from (select key from q1) a
+"""
+        )
+        == {Table("q2"), Table("src")}
+    )
```

Comment:
are we really sure that the `q2` reference in the `q1` CTE will in fact reference the table `q2` in the database? I did the following test on my devenv which has examples data and the superset metadata in the same database, and 

```sql
with q1 as ( select distinct gender from birth_names),
birth_names as ( select * from ab_role)
select * from (select gender from q1) a
```

produces the following error:

> sqlite error: no such column: gender

So to me it appears as if only `src` is in fact referenced here, right?

Reply:
I just tried something similar but it's definitely referencing the database table, not the alias which came after it:

```sql
postgres=# CREATE TABLE hodor(age INT, name VARCHAR);
CREATE TABLE
postgres=# INSERT INTO hodor(age, name) VALUES (35, 'Hodor');
INSERT 0 1
postgres=# CREATE TABLE potatoes(count INT);
CREATE TABLE
postgres=# INSERT INTO potatoes(count) VALUES (100);
INSERT 0 1
postgres=# WITH
  result AS (SELECT * FROM hodor),
  hodor AS (SELECT * FROM potatoes)
SELECT * FROM result;
 age | name  
-----+-------
  35 | Hodor
(1 row)
```

Reply:
Having said that, we're using different databases. It wouldn't surprise me at all if postgres and sqlite handled CTEs slightly differently. If that's the case this endeavour is slightly doomed though; we'd need a lot more logic here to handle various different databases.